modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/deep/pagination/configuration/DeepPaginationConfigurationWrapper.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.search.elasticsearch7.internal.deep.pagination.configuration;
+
+import com.liferay.portal.configuration.metatype.bnd.util.ConfigurableUtil;
+import com.liferay.portal.search.elasticsearch7.configuration.DeepPaginationConfiguration;
+
+import java.util.Map;
+
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+
+/**
+ * @author Gustavo Lima
+ */
+@Component(
+	configurationPid = "com.liferay.portal.search.elasticsearch7.configuration.DeepPaginationConfiguration",
+	service = DeepPaginationConfigurationWrapper.class
+)
+public class DeepPaginationConfigurationWrapper {
+
+	public boolean getEnableDeepPagination() {
+		return _deepPaginationConfiguration.enableDeepPagination();
+	}
+
+	public String getPointInTimeKeepAlive() {
+		return _validatePointInTimeString(
+			_deepPaginationConfiguration.pointInTimeKeepAlive());
+	}
+
+	@Activate
+	@Modified
+	protected void activate(Map<String, Object> map) {
+		_deepPaginationConfiguration = ConfigurableUtil.createConfigurable(
+			DeepPaginationConfiguration.class, map);
+	}
+
+	private String _validatePointInTimeString(String pointInTimeKeepAlive) {
```

Comment:
Not sure if this is good, because it seems that now faulty values can be stored and they are then silently ignored in the background.

 I think it would be better not being able to feed wrong values in the configuration at all. How about just changing the configuration to integer of seconds?

Reply:
looks good